page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js

```diff
@@ -4,65 +4,97 @@ const startPage = require('../generalPagesBeforeLogin/StartPageBeforeLogin');
+const addStudentSubmitBtn = 'div.modal.fade.add-modal.in button.btn-submit';
+const passwordInput = '#passwd';
 
-module.exports = {
-    createNewPupil: async function (firstname, lastname, email) {
-        await waitHelpers.waitAndClick(selectorAddStudentBtn);
-        await waitHelpers.waitAndSetValue(selectorSetFirstName, firstname);
-        await waitHelpers.waitAndSetValue(selectorSetLastName, lastname);
-        await waitHelpers.waitAndSetValue(selectorSetEmail, email);
-        await this.setStudentsBirthdayScript();
-        await waitHelpers.waitAndClick(selectorSendALinkBox);
-        await waitHelpers.waitAndClick(submitStudentCreateBtn);
-    },
-    setStudentsBirthdayScript: async function () {
-        await driver.pause(1500);
-        await driver.execute('document.querySelector("#create_birthday").value = "13.08.1990"')
-    },
+//
+async function clickAddStudentBtn() {
+	await elementHelpers.clickAndWait(addStudentBtn);
+}
+
+async function setStudentFirstName(firstname) {
+	await waitHelpers.waitAndSetValue(firstNameInput, firstname);
+}
+
+async function setStudentLastName(lastname) {
+	await waitHelpers.waitAndSetValue(lastNameInput, lastname);
+}
+
+async function setStudentEmail(email) {
+	await waitHelpers.waitAndSetValue(emailInput, email);
+}
+
+async function clickOnSendRegistrationLinkCheckbox() {
+	await elementHelpers.click(sendRegistrationLinkCheckbox);
+}
 
+async function submitStudentAddition() {
+	await elementHelpers.clickAndWait(addStudentSubmitBtn);
+}
 
-    emailsOfThePupils: async function () {
-        let names = await driver.$$(selectorNamesContainer + ' > tr');
-        return Promise.all(names.map(async (nameContainer) => {
-            const emailContainer = await nameContainer.$("td:nth-child(3)");
-            return await emailContainer.getText();
-        }))
-    },
-    verify: async function (email) {
-        let emails = await this.emailsOfThePupils();
-        await expect(emails).to.contain(email);
-    },
-    submitConsent: async function (e_mail) {
-        let names = await driver.$$(selectorNamesContainer + ' > tr');
-        for (var i = 1; i <= names.length; i++) {
-            let emailPromise = await driver.$(selectorNamesContainer + ' > tr:nth-child(' + i + ') > td:nth-child(3)');
-            let email = await emailPromise.getText();
-            if (email === e_mail) {
-                let boxConsent = await driver.$(selectorNamesContainer + ' > tr:nth-child(' + i + ') > td:nth-child(7) > a:nth-child(2) > i');
-                await boxConsent.click();
-                let submitBtn = await driver.$(selectorConsentSubmitBtn);
-                let passwordField = await driver.$(passwordFieldSel);
-                let password_old = await passwordField.getValue();
-                oldPassword = password_old;
-                await submitBtn.click();
-                break;
+async function createNewPupil(firstname, lastname, email) {
```

Comment:
this function looks very nice and clean :-)

Reply:
tnx :)